src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -116,15 +117,10 @@ export default class Pilot {
      * @return {array} [success of operation, readback]
      */
     maintainAltitude(altitude, expedite, shouldUseSoftCeiling, airportModel, aircraftModel) {
-        if (!aircraftModel.model.isAbleToMaintainAltitude(altitude)) {
-            const verbalRequestedAltitude = radio_altitude(altitude);
-            const readback = {};
-            readback.log = `unable to maintain ${altitude} due to performance`;
-            readback.say = `unable to maintain ${verbalRequestedAltitude} due to performance`;
-
-            this._mcp.setAltitudeFieldValue(aircraftModel.altitude);
+        const response = aircraftModel.validateNextAltitude(altitude);
 
-            return [false, readback];
+        if (response && !response[0]) {
```

Comment:
if the above method is update to always return an array, then this if becomes much easier:
```
if (response.length > 0) {
    return response;
}
```

Reply:
done